scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -918,6 +924,428 @@ cdef class TransformedDensityRejection(Method):
         return np.asarray(out).reshape(oshape)[()]
 
 
+UError = namedtuple('UError', ['max_error', 'max_absolute_error'])
+
+
+cdef class NumericalInversePolynomial(Method):
+    """
+    NumericalInversePolynomial(dist, mode=None, center=None, *, domain=None, order=5, u_resolution=1e-10, max_intervals=10000, keep_cdf=False, random_state=None)
+
+    Polynomial interpolation based INVersion of CDF (PINV).
+
+    PINV is a variant of numerical inversion, where the inverse CDF is approximated
+    using Newton's interpolating formula. The interval ``[0,1]`` is split into several
+    subintervals. In each of these, the inverse CDF is constructed at nodes ``(CDF(x),x)``
+    for some points ``x`` in this subinterval. If the PDF is given, then the CDF is
+    computed numerically from the given PDF using adaptive Gauss-Lobatto integration with
+    5 points. Subintervals are split until the requested accuracy goal is reached.
+
+    The method is not exact, as it only produces random variates of the approximated
+    distribution. Nevertheless, the maximal tolerated approximation error can be set to
+    be the resolution (but, of course, is bounded by the machine precision). We use the
+    u-error ``|U - CDF(X)|`` to measure the error where ``X`` is the approximate
+    percentile corressponding to the quantile ``U`` i.e. ``X = approx_ppf(U)``. We call
+    the maximal tolerated u-error the u-resolution of the algorithm in the sequel.
+
+    Both the order of the interpolating polynomial and the u-resolution can be selected.
+    Note that very small values of the u-resolution are possible but increase the cost
+    for the setup step.
+
+    The interpolating polynomials have to be computed in a setup step. However, it only
+    works for distributions with bounded domain; for distributions with unbounded domain
+    the tails are cut off such that the probability for the tail regions is small compared
+    to the given u-resolution.
+
+    The construction of the interpolation polynomial only works when the PDF is unimodal
+    or when the PDF does not vanish between two modes.
+
+    Parameters
+    ----------
+    dist : object
+        An instance of a class with a ``pdf`` and optionally a ``cdf`` method.
+
+        * ``pdf``: PDF of the distribution. The signature of the PDF is expected to be:
```

Comment:
we need to expain somewhere that there are restrictions on the pdf as in the UNU>RAN documentation>

There are some restrictions for the given distribution:

The support of the distribution (i.e., the region where the PDF is strictly positive) must be connected. In practice this means, that the region where PDF is "not too small" must be connected. Unimodal densities satisfy this condition. If this condition is violated then the domain of the distribution might be truncated.
When the PDF is integrated numerically, then the given PDF must be continuous and should be smooth.
The PDF must be bounded.
The algorithm has problems when the distribution has heavy tails (as then the inverse CDF becomes very steep at 0 or 1) and the requested u-resolution is very small. E.g., the Cauchy distribution is likely to show this problem when the requested u-resolution is less then 1.e-12.

Reply:
> we need to expain somewhere that there are restrictions on the pdf as in the UNU>RAN documentation>

I agree. I reserved that for the tutorial. Should we also mention it in the docs?

Reply:
I think we should since it is very important to know about the restrictions

Reply:
Done.